### DISCUSSION TITLE
Unable to install extension 'ms-toolsai.jupyter' as it is not compatible with VS Code version

### DISCUSSION THREAD

**Discussion by xavipolo** [microsoft/vscode-jupyter#10489] [Questions and Answers]
Hi,

I need to install the extension offline, I have downloaded the latest version (ms-toolsai.jupyter-2022.6.1001691033.vsix) and it gives me the following error: "Unable to install extension 'ms-toolsai.jupyter' as it is not compatible with VS Code '1.68.1'."

I have assumed that in the file numbering, the "16910033" part refers to the VS Code version, so I have tried with the "ms-toolsai.jupyter-2022.6.100**1681**045.vsix)" but I get the same error.

How can I install this extension offline?

Thanks

**Comment by xavipolo** [comment]
The solution for me has been to check in another VS Code 1.68.1 that has internet access, which version of the extension is installed (and updated).
Curiously, despite having VS Code updated, the version of the extension is "2022.5.1001601848" which is not the latest.

To download it, I have modified one of the download links that appear in the version history tab of the extension (https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter), with that particular version.

https://marketplace.visualstudio.com/_apis/public/gallery/publishers/ms-toolsai/vsextensions/jupyter/2022.5.1001601848/vspackage

This file can be installed in VS Code

easy, easy :-)

**Comment by nevdelap** [comment]
I have essentially the same problem with this extension always - how can one figure out what version is compatible with a version of VSCode? And how can one figure out what url to use to download it if none of the versions in the [marketplace](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter&ssr=false#version-history) are compatible? I'm on the April version because I've not bothered to figure it out recently, but having installed VSCode 1.82.2 I've tried to install those versions listed in the marketplace with no luck. 🙏

**Comment by DonJayamanne** [comment]
Duplicate https://github.com/microsoft/vscode/issues/192738
